src/commands/ontology.js

```diff
@@ -111,6 +113,35 @@ class VocabLoad extends Command {
   }
 }
 
+class VocabExport extends Command {
+  static get action() { return VOCAB.EXPORT }
+
+  *exec() {
+    const { payload } = this.action
+    const [vocab, classes, props, types] =
+          yield select(state => [
+            pluck(state.ontology.vocab, payload),
+            state.ontology.class,
+            state.ontology.props,
+            state.ontology.type
+          ])
+
+    const path = yield call(dialog.save.vocab)
+    if (!path) return
+
+    const data = yield call(toN3, vocab[0], classes, props, types)
+
+    try {
+      yield call(async () => { write(path, await data) })
```

Comment:
This seems odd to me: why do we need to `await` data here? If `toN3` returns a promise then `yield call(toN3, ...)` will suspend the generator until the promise is resolved (or throw an error inside when it is rejected).

We should also remove the try-catch here I think. I currently guards only against errors when writing the file (a promise rejection in `toN3` would throw outside the try block). But I don't think we need to catch errors here at all: by catching the error (and not throwing it again) the command is marked as successful; if we don't catch the error here then it would be caught in the `Command` superclass and the command marked as failed; this would cause the error to be logged and the failure dialog to pop up. We usually catch errors inside commands only if we can handle them or if the error does not mean the whole command should fail (e.g., when importing 100 items, individual items can fail, but the import command would still run to completion).

Reply:
I see, that simplifies it (I forgot that `yield call` will resolve a potential Promise)